controllers/databaseclusterbackup_controller.go

```diff
@@ -372,10 +373,21 @@ func (r *DatabaseClusterBackupReconciler) tryCreatePG(ctx context.Context, obj c
 		databaseClusterNameLabel:                      pgBackup.Spec.PGCluster,
 		fmt.Sprintf(backupStorageNameLabelTmpl, name): backupStorageLabelValue,
 	}
+	backup.SetOwnerReferences(dbOwnerRefMeta(cluster))
 
 	return r.Create(ctx, backup)
 }
 
+func dbOwnerRefMeta(db *everestv1alpha1.DatabaseCluster) []metav1.OwnerReference {
+	return []metav1.OwnerReference{{
+		APIVersion:         everestAPIVersion,
+		Kind:               databaseClusterKind,
+		Name:               db.Name,
+		UID:                db.UID,
+		BlockOwnerDeletion: pointer.ToBool(true),
+	}}
+}
+
```

Comment:
We can use `controllerutil.SetControllerReference` directly, then we can remove this helper function. See example: https://github.com/percona/everest-operator/blob/main/controllers/monitoringconfig_controller.go#L96

Reply:
Thanks, updated